pdp/static/js/pdp_controls.js

```diff
@@ -181,26 +181,40 @@ function generateMenuTree(subtree, leafNameMapping) {
             leafNameMapping[b] : b).toLowerCase();
         return adn.localeCompare(bdn);
     }
-
-    $.each(Object.keys(subtree).sort(compareDisplayName), function (index, stuff) {
-        var newlayer, linkText,
-            li = $('<li/>');
-        if (subtree[stuff] instanceof Object) {
-            li.append($('<a/>').text(stuff)).append(generateMenuTree(subtree[stuff], leafNameMapping));
-        } else {
-            newlayer = subtree[stuff] + "/" + stuff;
-            linkText = stuff;
-            if (leafNameMapping !== undefined) {
-                linkText = leafNameMapping[stuff];
+    if (pcic12) { // Arrange PCIC12 models in specific order
+        const models = ["BCC-CSM2-MR", "NorESM2-LM", "MIROC-ES2L",
+        "MPI-ESM1-2-HR", "MRI-ESM2-0", "UKESM1-0-LL", "EC-Earth3-Veg",
+        "CMCC-ESM2", "INM-CM5-0", "FGOALS-g3", "TaiESM1", "IPSL-CM6A-LR"];
+        models.forEach((model) => {
+            var li = $('<li/>');
+            li.append($('<a/>').text(model)).append(generateMenuTree(subtree[model], leafNameMapping));
+            li.appendTo(ul);
+        });
+    } else {
+        $.each(Object.keys(subtree).sort(compareDisplayName), function (index, stuff) {
+            var newlayer, linkText,
+                li = $('<li/>');
+            if (subtree[stuff] instanceof Object) {
+                if (stuff.includes("PCIC12")) {
+                    li.append($('<a/>').text(stuff)).append(generateMenuTree(subtree[stuff], leafNameMapping, true));
+                } else {
+                    li.append($('<a/>').text(stuff)).append(generateMenuTree(subtree[stuff], leafNameMapping));
+                }
+            } else {
+                newlayer = subtree[stuff] + "/" + stuff;
+                linkText = stuff;
+                if (leafNameMapping !== undefined) {
+                    linkText = leafNameMapping[stuff];
+                }
+                li.attr('id', newlayer);
+                $('<a/>').text(linkText).click(function () {
+                    ncwms.params.LAYERS = newlayer;
+                    ncwms.events.triggerEvent('change', newlayer);
+                }).addClass('menu-leaf').appendTo(li);
             }
-            li.attr('id', newlayer);
-            $('<a/>').text(linkText).click(function () {
-                ncwms.params.LAYERS = newlayer;
-                ncwms.events.triggerEvent('change', newlayer);
-            }).addClass('menu-leaf').appendTo(li);
-        }
-        li.appendTo(ul);
-    });
+            li.appendTo(ul);
+        });
+    }
```

Comment:
This was always a complicated block of code, and it's more complicated now. It's hard to understand. We should consider breaking it down into two or more functions. An `li`-maker or two would be a place to start.

Reply:
Much better. Still a bit complicated as you noted, but it can wait.